addons/turbowarp-player/userscript.js

```diff
@@ -27,12 +27,24 @@ export default async function ({ addon, console, msg }) {
   }
 
   button.onclick = async () => {
+    const projectId = window.location.pathname.split("/")[2];
     let search = "";
+    let fetchHeaders = {};
+    if (await addon.auth.fetchXToken()) {
+      fetchHeaders["x-token"] = await addon.auth.fetchXToken();
+    }
+    let projectToken = (
+      await (
+        await fetch(`https://api.scratch.mit.edu/projects/${projectId}?nocache=${Date.now()}`, {
+          headers: fetchHeaders,
+        })
+      ).json()
+    ).project_token;
     if (
       addon.tab.redux.state?.preview?.projectInfo?.public === false &&
```

Comment:
if the project is shared then don't even bother trying to fetch a new token, the code just ignores it anyways

Reply:
Alright, I fixed it to only fetch the token if the project isn't public. Otherwise, it just replaces the project player normally.